include/odp/api/spec/schedule_types.h

```diff
@@ -159,6 +165,109 @@ typedef	struct odp_schedule_param_t {
 	uint32_t lock_count;
 } odp_schedule_param_t;
 
+/**
+ * Scheduler capabilities
+ */
+typedef struct odp_schedule_capability_t {
+	/** Maximum number of ordered locks per queue */
+	uint32_t max_ordered_locks;
+
+	/** Maximum number of scheduling groups */
+	uint32_t max_groups;
+
+	/** Number of scheduling priorities */
+	uint32_t max_prios;
+
+	/** Scheduled queue capabilities */
+	struct {
+		/** Maximum number of scheduled (ODP_BLOCKING) queues of the
+		  * default size. */
+		uint32_t max_num;
+
+		/** Maximum number of events a scheduled (ODP_BLOCKING) queue
+		  * can store simultaneously. The value of zero means that
+		  * scheduled queues do not have a size limit, but a single
+		  * queue can store all available events. */
+		uint32_t max_size;
+
+		/** Lock-free (ODP_NONBLOCKING_LF) implementation capabilities.
+		  * The specification is the same as for the blocking
+		  * implementation. */
+		struct {
+			/** Maximum number of queues. Lock-free queues are not
+			  * supported when zero. */
+			uint32_t max_num;
+
+			/** Maximum queue size */
+			uint32_t max_size;
+
+		} lockfree;
+
+		/** Wait-free (ODP_NONBLOCKING_WF) implementation capabilities.
+		  * The specification is the same as for the blocking
+		  * implementation. */
+		struct {
+			/** Maximum number of queues. Wait-free queues are not
+			  * supported when zero. */
+			uint32_t max_num;
+
+			/** Maximum queue size */
+			uint32_t max_size;
+
+		} waitfree;
+
+		/** Maximum supported flows per queue.
+		 * Specifies the maximum number of flows per queue supported by
+		 * the implementation.  A value of 0 indicates flow aware mode
+		 * is not supported.
+		 */
+		uint32_t max_flows;
+
+	} queue;
+
+} odp_schedule_capability_t;
```

Comment:
As per config_t comment: structs queue/lockfree/waitfree can be removed. So, that capa would be simply

typedef struct odp_schedule_capability_t {
uint32_t max_ordered_locks;
uint32_t max_groups;
uint32_t max_prios;
uint32_t max_queues;
uint32_t max_queue_size;
uint32_t max_flows;
odp_bool_t lockfree_queues;
odp_bool_t waitfree_queues;
} odp_schedule_capability_t;

Reply:
I hoped to retain close compatibility between old queue capa and scheduler capa.